lib/ansible/modules/cloud/vmware/vmware_dvs_portgroup.py

```diff
@@ -157,242 +217,1038 @@
+            if vendor_config_override:
+                results['adv_vendor_conf_previous'] = pg_config.policy.vendorConfigOverrideAllowed
+            if vlan_override:
+                results['adv_vlan_previous'] = pg_config.policy.vlanOverrideAllowed
+            if uplink_teaming_override:
+                results['adv_uplink_teaming_previous'] = pg_config.policy.uplinkTeamingOverrideAllowed
+            if network_rp_override:
+                results['adv_network_rp_previous'] = pg_config.policy.networkResourcePoolOverrideAllowed
+            if security_override:
+                results['adv_security_previous'] = pg_config.policy.securityPolicyOverrideAllowed
+            if netflow_override:
+                results['adv_netflow_previous'] = pg_config.policy.ipfixOverrideAllowed
+            if traffic_filter_override:
+                results['adv_traffic_filtering_previous'] = pg_config.policy.trafficFilterOverrideAllowed
+            pg_spec.policy = pg_policy_spec
+
+        pg_spec.defaultPortConfig = vim.dvs.VmwareDistributedVirtualSwitch.VmwarePortConfigPolicy()
+
+        # TODO: Check Traffic shaping
+
+        # Check VLAN
+        if self.vlan_id == 0 and self.vlan_trunk_range is None and self.private_vlan_id is None:
+            results['vlan_id'] = 0
+            if isinstance(pg_config.defaultPortConfig.vlan, vim.dvs.VmwareDistributedVirtualSwitch.VlanIdSpec):
+                if pg_config.defaultPortConfig.vlan.vlanId != 0:
+                    changed_vlan = True
+            elif isinstance(pg_config.defaultPortConfig.vlan, vim.dvs.VmwareDistributedVirtualSwitch.TrunkVlanSpec):
+                changed_vlan = True
+                results['vlan_trunk_range_previous'] = self.get_current_trunk_vlan_range(pg_config)
+            elif isinstance(pg_config.defaultPortConfig.vlan, vim.dvs.VmwareDistributedVirtualSwitch.PvlanSpec):
+                changed_vlan = True
+                results['private_vlan_id_previous'] = pg_config.defaultPortConfig.vlan.pvlanId
+            if changed_vlan:
+                changed = True
+                changed_list.append("vlan")
+                pg_spec.defaultPortConfig.vlan = vim.dvs.VmwareDistributedVirtualSwitch.VlanIdSpec()
+                pg_spec.defaultPortConfig.vlan.vlanId = 0
+        # NOTE: 'vlan_trunk' option is deprecated. Will be removed in 2.11
+        elif self.vlan_trunk or self.vlan_trunk_range:
+            results['vlan_trunk_range'] = self.vlan_trunk_range
+            if isinstance(pg_config.defaultPortConfig.vlan, vim.dvs.VmwareDistributedVirtualSwitch.VlanIdSpec):
+                changed_vlan = True
+                results['vlan_id_previous'] = pg_config.defaultPortConfig.vlan.vlanId
+            elif isinstance(pg_config.defaultPortConfig.vlan, vim.dvs.VmwareDistributedVirtualSwitch.TrunkVlanSpec):
+                changed_vlan_trunk_range, trunk_vlan_spec = self.check_trunk_vlan_config(pg_config)
```

Comment:
if you change `check_trunk_vlan_config` to return in the same order as the others, the variables here need to be reversed (and any `result[1]` in the code, needs to change to `result[0]`)

Reply:
done. thx.